src/components/cms-components/BlogsList/BlogsList.astro

```diff
@@ -0,0 +1,363 @@
+---
+import { blogsListQuery } from "@api/queries";
+import { imageBuilder, loadQuery } from "@api/sanity";
+import type {
+  BlogPostWithReferences,
+  BlogsListReferenced,
+} from "@sanity/custom.sanity.types";
+import { isPreview } from "src/stores/global-store";
+import { toHTML } from "@portabletext/to-html";
+import { v4 as uuidv4 } from "uuid";
+import { getLocale } from "i18n:astro";
+import { t } from "i18next";
+
+interface Props {
+  blogsList: BlogsListReferenced;
+}
+const itemsPerTime = 3;
+const { blogsList } = Astro.props;
+
+let blogs: BlogPostWithReferences[] | undefined = [];
+let totalItems;
+if (blogsList.blogsType === "latest") {
+  const { data: blogsFetched } = await loadQuery<{
+    blogPosts: BlogPostWithReferences[];
+    totalItems: number;
+  }>({
+    query: blogsListQuery,
+    params: { language: getLocale(), itemsPerTime: itemsPerTime },
+  });
+  blogs = blogsFetched.blogPosts;
+  totalItems = blogsFetched.totalItems;
+} else {
+  blogs = blogsList.blogPosts;
+  totalItems = blogs?.length;
+}
+
+// UUID for giving the divs and buttons unique ID's we can select them in JS
+const uuid = uuidv4();
+
+// If we are fetching the newest blogs, or if we are using specific blogs and no fetching is needed then.
+const isLatestBlogs = blogsList.blogsType === "latest";
+// If we are fetching the latestblogs, give the amount of items from the api which can be max itemsPerTime
+// If we are fetching specific blogs, give the lenght of the blogs, with a maximum of the itemsPerTime.
+const items = isLatestBlogs
+  ? blogs?.length
+  : blogs
+    ? blogs?.length > itemsPerTime
+      ? itemsPerTime
+      : blogs?.length
+    : 0;
+
+const preview = isPreview.get();
+---
+
+<script>
+  await import.meta.glob("../../webcomponents/*.js", {
+    eager: true,
+  });
+</script>
+
+{
+  blogs && (
+    <astro-blogs-list
```

Comment:
I feel like this can all be simplified if we reuse the already existing BlogsPreview: https://github.com/amexio-belgium/cxm-minisite/blob/11c4eaa3cd28e9ca3ec5893d7801d28781f9a79b/src/pages/%5Blang%5D/insights/index.astro
This seems closer to the implementation for services and works as well